core/src/runtime/syscall.rs

```diff
@@ -47,6 +51,7 @@ impl SyscallCode {
     match value {
       0x00_00_00_00 => SyscallCode::HALT,
       0x00_00_00_02 => SyscallCode::WRITE,
+      0x00_64_00_20 => SyscallCode::PRECOMPILE_ED25519_VERIFY,
```

Comment:
let's keep these in order... there will be more precompiles, let's group them together below the existing sys calls

Reply:
What do you mean? They are ordered by syscall ID. Btw, I realized that each byte of the syscall code has a special meaning (hence 0x64) for 100 gas. I'm not sure if we want to keep it. It's weird to have gas cost encoded in the syscall code.

Reply:
I see. Yeah, afaict this is NOT dictated by the RISC-V spec itself, it seems to be something we carried over from SP1. And it's not enforced, other than with a test. I think we can drop it, but let's do so under a separate PR.

I can't see any benefit to encoding this in the syscall ID, can you?